Patch-Source: https://github.com/AcademySoftwareFoundation/OpenShadingLanguage/commit/9cfca9397b974f00bcc0915a4661be19e2e6e820
From 9cfca9397b974f00bcc0915a4661be19e2e6e820 Mon Sep 17 00:00:00 2001
From: Larry Gritz <lg@larrygritz.com>
Date: Wed, 27 Apr 2022 23:55:08 -0700
Subject: [PATCH] Support for LLVM 14 (#1492)

API changes we had to take into account:
* TargetRegistry.h location
* No more DisableTailCalls field in PassManagerBuilder.

Needed to update the ref image for render-microfacet test, some sparklies changed.
Looks like the new LLVM probably JITs to ever so slightly different math code, tickling
some LSB differences that at 1 sample per pixel, results in some different sampling
directions leading to fireflies. We decided to just commit a new ref image and move on.

Signed-off-by: Larry Gritz <lg@larrygritz.com>
---
 .github/workflows/ci.yml                       |  10 +++++-----
 INSTALL.md                                     |   2 +-
 src/build-scripts/build_llvm.bash              |   2 +-
 src/liboslexec/llvm_util.cpp                   |   6 ++++--
 testsuite/render-microfacet/ref/out-llvm14.exr | Bin 0 -> 332117 bytes
 5 files changed, 11 insertions(+), 9 deletions(-)
 create mode 100644 testsuite/render-microfacet/ref/out-llvm14.exr

diff --git a/src/liboslexec/llvm_util.cpp b/src/liboslexec/llvm_util.cpp
index b30c89318..a917d621e 100644
--- a/src/liboslexec/llvm_util.cpp
+++ b/src/liboslexec/llvm_util.cpp
@@ -45,7 +45,11 @@
 #include <llvm/Support/raw_os_ostream.h>
 #include <llvm/IR/LegacyPassManager.h>
 #include <llvm/IR/ValueSymbolTable.h>
+#if OSL_LLVM_VERSION < 140
 #include <llvm/Support/TargetRegistry.h>
+#else
+#include <llvm/MC/TargetRegistry.h>
+#endif
 
 #include <llvm/Bitcode/BitcodeReader.h>
 #include <llvm/Bitcode/BitcodeWriter.h>
